src/System.IO.IsolatedStorage/src/System/IO/IsolatedStorage/Helper.Unix.cs

```diff
@@ -0,0 +1,17 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+namespace System.IO.IsolatedStorage
+{
+    internal static partial class Helper
+    {
+        internal static void CreateDirectory(string path, IsolatedStorageScope scope)
+        {
+            if (Directory.Exists(path))
```

Comment:
What happens if there's a race condition and the directory is created between the Exists check and the subsequent CreateDirectory call... will CreateDirectory throw an exception?  If so, should we eat it?